### DISCUSSION TITLE
I want to make block rather than rectangular box.

### DISCUSSION THREAD

**Discussion by bosxered** [idaholab/moose#24876] [Q&A Meshing] [answered]
Hello Moose experts.

I want to make a simulation block like the following.

**1. Trapezoid.**

![image](https://github.com/idaholab/moose/assets/79071957/6c5cb267-e0aa-4da6-89bb-ad28f347367a)
Figure 1.

I wanted to make a trapezoidal block.
I tried StackGenerator and combined rectangular block and triangular block but it failed.
Here is the input file and the error message.

```
#####
xmax = 35
ymax = 20
#####

[Mesh]
  [./poly]
    type = PolyLineMeshGenerator
    points = '0.0 0.0 0.0
              ${xmax} 2.0 0.0
              ${xmax} 0.0 0.0'
    loop = true
#   num_edges_between_points = 20
  []
  [./tri]
    type = XYDelaunayGenerator
    boundary = 'poly'
    refine_boundary = false
    desired_area = 1
#    output = true
  []
  [./gen]
    type = GeneratedMeshGenerator
    dim = 2
    nx = 35
    ny = 20
    nz = 0
    xmin = 0.00
    xmax = ${xmax}
    ymin = 0.00
    ymax = ${ymax}
    zmin = 0
    zmax = 0
    elem_type = QUAD4
  [../]
  [./stack]
    type = StackGenerator
    dim = 2
    inputs = 'gen tri'
  []
[]
```

```
===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   RANK 0 PID 72460 RUNNING AT cnode14
=   KILLED BY SIGNAL: 11 (Segmentation fault)
===================================================================================
```

The reason why I didn't use only the PolyLineMeshGenerator and the XYDelaunayGenerator is that I believed I didn't want to make uneven and rough initial condition like Figure 2. An example of even and clean-cut initial condition is shown in Figure 3.

![image](https://github.com/idaholab/moose/assets/79071957/5bc5f17d-0d58-45ea-9c05-7bace6e5fc68)
![image](https://github.com/idaholab/moose/assets/79071957/fac7b156-40cf-456a-8f67-e180062adf26)
Figure 2. When I used only the PolyLineMeshGenerator and the XYDelaunayGenerator. Uneven and rough initial condition are made.

![image](https://github.com/idaholab/moose/assets/79071957/033454c2-193c-4e81-b524-62592e02ed9c)
Figure 3. An example of even and clean-cut initial condition.

**2. Boundary can be expressed as a simple function of x.**

Also, I want to make a block of which boundary can be expressed as a function of x such as sine function or polynomial.

![image](https://github.com/idaholab/moose/assets/79071957/7638b5d8-0fa5-4b5e-87b3-1965e28e343b)
Figure 4.

Is there any way to make this? Can you tell me how?

Thank you so much for reading.

**Comment by GiudGiud** [comment, marked as answer]
Hello

you are on the right track by making parsed curves. 

You just want to not use the stack generator and use a stitchedmeshgenerator instead to merge the two blocks. 

Also please note that you cannot have BOTH quad4 and tri elements in the same subdomain, they need to be in different subdomains (but they can be in the same mesh)

Guillaume

**Reply by bosxered** [reply]
Thank you so much!
